Assets/LeapMotion/Core/Scripts/Utils/Utils.cs

```diff
@@ -998,168 +932,108 @@ public struct ChildrenEnumerator : IEnumerator<Transform> {
       return new Quaternion(x / magnitude, y / magnitude, z / magnitude, w / magnitude);
     }
 
-    #endregion
-
-    #region Float Utils
-
     /// <summary>
-    /// Additive From syntax for floats. Evaluated as this float plus the additive
-    /// inverse of the other float, usually expressed as thisFloat - otherFloat.
+    /// Returns the rotation that makes a transform at fromPosition point its forward
+    /// vector at targetPosition and keep its rightward vector parallel with the horizon
+    /// defined by a normal of Vector3.up.
     /// 
-    /// For less trivial uses of From/Then syntax, refer to their implementations for
-    /// Quaternions and Matrix4x4s.
+    /// For example, this will point an interface panel at a user camera while
+    /// maintaining the alignment of text and other elements with the horizon line.
     /// </summary>
-    public static float From(this float thisFloat, float otherFloat) {
-      return thisFloat - otherFloat;
+    /// <returns></returns>
+    public static Quaternion FaceTargetWithoutTwist(Vector3 fromPosition,
+                                                    Vector3 targetPosition,
+                                                    bool flip180 = false) {
+      return FaceTargetWithoutTwist(fromPosition, targetPosition, Vector3.up, flip180);
     }
 
     /// <summary>
-    /// Additive To syntax for floats. Evaluated as this float plus the additive
-    /// inverse of the other float, usually expressed as otherFloat - thisFloat.
+    /// Returns the rotation that makes a transform at fromPosition point its forward
+    /// vector at targetPosition and keep its rightward vector parallel with the horizon
+    /// defined by the upwardDirection normal.
     /// 
-    /// For less trivial uses of From/Then syntax, refer to their implementations for
-    /// Quaternions and Matrix4x4s.
+    /// For example, this will point an interface panel at a user camera while
+    /// maintaining the alignment of text and other elements with the horizon line.
     /// </summary>
-    public static float To(this float thisFloat, float otherFloat) {
-      return otherFloat - thisFloat;
+    public static Quaternion FaceTargetWithoutTwist(Vector3 fromPosition,
+                                                    Vector3 targetPosition,
+                                                    Vector3 upwardDirection,
+                                                    bool flip180 = false) {
+      Vector3 objToTarget = targetPosition - fromPosition;
+      return Quaternion.LookRotation((flip180 ? -1 : 1) * objToTarget,
+                                     upwardDirection);
     }
 
-    /// <summary>
-    /// Additive Then syntax for floats. Literally, thisFloat + otherFloat.
-    /// </summary>
-    public static float Then(this float thisFloat, float otherFloat) {
-      return thisFloat + otherFloat;
+    public static Vector3 GetRight(this Quaternion q) {
```

Comment:
I'm not convinced that we need dedicated extension methods on a Quaternion to do a rotation like this.  I think it would need a more details name to be obvious what it actually does, by which time you might as well be writing the original code in the first place.